src/main/java/ru/hogwarts/school/service/FacultyService.java

```diff
@@ -20,26 +20,38 @@ public FacultyService (FacultyRepository facultyRepository)
 
     public Faculty addFaculty(Faculty faculty)
     {
+        if(faculty == null)
+            return null;
+
         facultyRepository.save(faculty);
         return faculty;
     }
 
-    public Faculty getFaculty(long id)
+    public Faculty getFaculty(Long id)
     {
+        if(facultyRepository.findById(id).isEmpty())
+            return null;
         return facultyRepository.findById(id).get();
```

Comment:
```suggestion
        return facultyRepository.findById(id).orElse(null);
```
вот так надо было обработать, а не с проверкой через if